### PR TITLE
Move JSONPb into its own package

### LINKED ISSUE #10294: Using your workaround for grpc-gateway and gogo/protobuf inconsistencies is inefficient.
This is a simple issue, I tried vendoring https://github.com/cockroachdb/cockroach/blob/master/pkg/util/jsonpb_marshal.go and because it's part of the larger `util` package rather than it's own self-contained sub-package, as it could be (and maybe should be?), I get the whole util package with all it's sub-dependencies, when I just want your customer marshaler. It would be super nice if the marshaler could be moved to a standalone subpackage so it's possible to just import that on it's own.

### PR DESCRIPTION
**Pull request by johanbrandhorst**
Fixes #10294.

---

This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/10295)

### CODE REVIEW

File: pkg/util/pbmarshal/proto_marshal.go [hunk 1]
```diff
@@ -34,7 +34,7 @@ type ProtoPb struct{}
 
 // ContentType implements gwruntime.Marshaler.
 func (*ProtoPb) ContentType() string {
-	return ProtoContentType
+	return "application/x-protobuf"
```
Comment (johanbrandhorst) [inline pkg/util/pbmarshal/proto_marshal.go]: Note that this is consistent with jsonpb_marshal.go already

**Comment by johanbrandhorst** [comment]
Currently battling with a few options here with regards to the string constants. I would like to have ProtoContentType and the others in the `pbmarshal` package, but then it wouldn't be in the same place as the other content types, which are in `utils/http.go`. If they're not moved, this new package is either forced to import utils, which makes this change pointless, or just return string literals, like it is now. Opinions very much welcome.

**Comment by tamird** [comment]
Hi @johanbrandhorst, thanks for the pull request. Couple things:

1) We already have a package called protoutil, perhaps this can go in there?
2) I would prefer that we avoid string literals. If we already have one in jsonpb_marshal.go, we should fix it, rather than add more. If you need to break dependencies, you could make a new httputil package containing `util/http.go`
3) Please squash all your commits down; we don't want to pollute the git history with so much "iteration".

**Comment by johanbrandhorst** [comment]
@tamird thanks, I'll give it another iteration and squash it once you think it looks good!

File: pkg/util/protoutil/marshal.go [hunk 2]
```diff
@@ -1,31 +0,0 @@
...
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
-// implied. See the License for the specific language governing
-// permissions and limitations under the License.
-//
-// Author: Tamir Duberstein (tamird@gmail.com)
-
-package protoutil
```
Comment (tamird) [inline pkg/util/protoutil/marshal.go]: leave this stuff in this file

File: pkg/util/protoutil/proto_marshal.go [hunk 3]
```diff
@@ -14,7 +14,7 @@
 //
 // Author: Tamir Duberstein (tamird@gmail.com)
 
-package util
+package protoutil
```
Comment (tamird) [inline pkg/util/protoutil/proto_marshal.go]: name this file marshaler.go, if you don't mind.

**Review by tamird — COMMENTED** [review]
looking good!

File: pkg/util/protoutil/marshal.go (hunk 2 above)
Comment (johanbrandhorst) [inline pkg/util/protoutil/marshal.go]: Recreated file

**Comment by johanbrandhorst** [comment]
Squashed everything up as well. Anything else?

**Comment by tamird** [comment]
Thanks!

**Comment by johanbrandhorst** [comment]
Thank you!

**Pull request merged**